Source/WebCore/editing/ReplaceSelectionCommand.cpp

```diff
@@ -531,9 +533,13 @@ static bool isHeaderElement(const Node& a)
         || a.hasTagName(h6Tag);
 }
 
-static bool haveSameTagName(Node* a, Node* b)
+static bool haveSameTagName(Node& a, RefPtr<Node> b)
```

Comment:
It strikes me as strange that one of them is a reference and the other one is `RefPtr`.

Reply:
I could have added a null check to the caller. I suggest we revisit this once we locate more potential callers for this. Some light grepping doesn't reveal any obvious pattern and it seems a bit suspect to compare the tag name (rather than local name) to begin with.

Reply:
@annevk Could you please revert this change? It is not OK to pass a RefPtr<> like this as parameter. This causes ref-counting churn.

Reply:
In case anyone reads this in the future, #23106 fixes this.